test/Makefile

```
@@ -25,6 +29,110 @@ clean: # Cleanup the test environment.
 .PHONY: test
 test: # Run end-to-end tests.
 
+integration-test:
+	cd $(ROOT_DIR); go test -count=1 -tags=integration ./...
+
+e2e-test: clean-test-cluster test-cluster _e2e-test
+	$(MAKE) clean-test-cluster
+
+_e2e-test:
+	$(CHAINSAW) test --parallel 1 --test-dir $(ROOT_DIR)/test/e2e --selector $(TEST_SELECTOR)
+
+test-cluster: ctlptl kind helm
+	$(CTLPTL) apply -f $(ROOT_DIR)/test/ctlptl-config.yaml
+	kubectl create -k github.com/kubernetes-sigs/container-object-storage-interface-api
+	kubectl create -k github.com/kubernetes-sigs/container-object-storage-interface-controller
+	$(HELM) install linode-cosi-driver \
+        $(ROOT_DIR)/helm/linode-cosi-driver \
+        --set=apiToken=$(LINODE_TOKEN) \
```

```suggestion
        --set=apiToken=$(LINODE_TOKEN) \
        --set=driver.image.tag=$(VERSION) \
```

Also adding the following to the Makefile?

```
VERSION ?= $(shell git tag | tail -n 1 | grep '' || echo 'v0.0.0')$(shell git diff --quiet || git rev-parse HEAD | sed 's/\(.\{6\}\).*/-\1/')
```

---

Do you have an idea for loading the image into KinD's registry before installing the chart here?

---

@shanduur im workong on it.